packages/overlay/src/overlay-stack.ts

```diff
@@ -611,9 +611,11 @@ export class OverlayStack {
         while (index && overlaysToClose.length === 0) {
             index -= 1;
             const overlay = this.overlays[index];
+            const path = event.composedPath();
```

Comment:
I may not understand what's being checked here enough to label it properly, but consider something like this?

```suggestion
            const path = event.composedPath();
            const isOutsideTriggerAndContent = (!path.includes(overlay.trigger) || overlay.interaction !== 'hover') && !path.includes(overlay.overlayContent;
```

Reply:
Good call. The naming here is hard, but I've paired it with an update comment from the change that broke this, so hopefully it's more understandable in the future.